bot/voice.go

```diff
@@ -0,0 +1,99 @@
+package bot
+
+import (
+	"log"
+	"strings"
+	"time"
+
+	"github.com/bwmarrin/discordgo"
+)
+
+var activeReceiver *AudioReceiver
+
+func stopActiveReceiver() {
+	if activeReceiver != nil {
+		activeReceiver.Stop()
+		activeReceiver = nil
+	}
+}
+
+func dcCmd(ctx CommandCtx) error {
+	if ctx.Message.Author.ID == morthisId {
```

Comment:
I'm begging you to break this habit. Whenever you see _any_ huge indented block like this (but especially when we are inside a function with such easy control flow manipulation via `return`) its a red flag that something is wrong or can be simplified. Invert this check and use a return so the majority of the code can remain at a single indentation level.

```go
if ctx.Message.Author.ID != morthisId {
        return ctx.Reply("stinky")
}
```